unit_tests/unit/test_scylla_yaml_builders.py

```
@@ -58,326 +64,320 @@ class FakeNode:
-class ScyllaYamlClusterAttrBuilderBase:
-    builder_class: ScyllaYamlAttrBuilderBase
-
-    def _run_test(self, builder_params: dict, expected_as_dict: dict = None):
-        instance = self.builder_class(**builder_params)
-        resulted_as_dict = instance.model_dump(exclude_defaults=True, exclude_unset=True)
-        if expected_as_dict:
-            assert resulted_as_dict == expected_as_dict
-            ScyllaYaml(**resulted_as_dict)
-
+def assert_builder_output(
+    builder_class: ScyllaYamlAttrBuilderBase, builder_params: dict, expected_as_dict: dict = None
```

Ruff (0.15.21)

[warning] PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)